server/src/organizations/organizations.service.ts

```diff
@@ -1,19 +1,45 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
 import { CreateOrganizationDto } from './dto/create-organization.dto';
 import { UpdateOrganizationDto } from './dto/update-organization.dto';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository, DeleteResult } from 'typeorm';
 import { Organization } from './entities/organization.entity';
+import fetch from 'node-fetch';
+
+export type EINCheck = {
+  validEIN: boolean;
+  proPublicaName: string;
+};
 
 @Injectable()
 export class OrganizationsService {
   constructor(
-    @InjectRepository(Organization)
-    private organizationsRepository: Repository<Organization>,
+    @InjectRepository(Organization) private organizationsRepository: Repository<Organization>,
   ) {}
 
   async create(createOrganizationDto: CreateOrganizationDto): Promise<Organization> {
-    return await this.organizationsRepository.save(createOrganizationDto);
+    return this.organizationsRepository.save(createOrganizationDto);
+  }
+
+  async checkEIN(ein: number): Promise<EINCheck> {
+    const einObj = {
+      validEIN: false,
+      proPublicaName: '',
+    };
+
+    try {
+      const res = await fetch(
+        `https://projects.propublica.org/nonprofits/api/v2/organizations/${ein}.json`,
+      );
+      const org = await res.json();
+      if (org) {
+        einObj.validEIN = true;
+        einObj.proPublicaName = org.organization.name;
+      }
+    } catch (err) {
+      console.log(err.message);
+    }
+    return einObj;
```

Comment:
let's just return the entire propublica org or null
and let's call this function, and the controller function, getPropublicaOrg or something

Reply:
Should I create a type that covers all properties of the propublica org? Or is there a TS hack I don't know about that allows you to return a generic object? Just trying to figure out what the return type will be.

Reply:
...ah just found this: https://www.typescriptlang.org/docs/handbook/2/generics.html

Reply:
Nevermind, I don't know how helpful that is. Will play around.

Reply:
Okay i'm an idiot—disregard!